_source/_posts/2024-08-15-otp-over-sms.md

```diff
@@ -0,0 +1,178 @@
+
+* Generating codes in an authenticator app such as Okta Verify, Authy, Google Authenticator, or 1Password.
+
+* FIDO2.0 (WebAuthn) which, in addition to phones, can use hardware keys and on-device authenticators.
+
+Soon, [Okta will](https://support.okta.com/help/s/article/bring-your-own-telephony-required-for-sms-and-voice?language=en_US)[ require you to bring your own telephony provider](https://support.okta.com/help/s/article/bring-your-own-telephony-required-for-sms-and-voice?language=en_US) to keep sending those codes. If you need time to move to a different method of verifying identity, you must configure your own provider for SMS/Voice.
+
+## Hooked on telephony
+
+You can send the OTP in the SMS/Voice flow using the [telephony inline hook]([https://help.okta.com/oie/en-us/content/topics/telephony/telephony-inline-hook.htm](https://help.okta.com/oie/en-us/content/topics/telephony/telephony-inline-hook.htm)). Okta uses the code or URL in the hook to send the OTP, though, as you'll see, the hook may not be called every time (and that's a good thing). When your hook fails to send the message or takes too long to update the status, Okta takes over sending the message. However, the number of those messages is heavily rate-limited.
+
+The code or URL you provide may simply send the message and communicate the outcome to Okta. The code or server may be more complex, managing geo-specific vendors, failure, failover to another provider, and hacking. No matter how easy or complex the code, there are three main approaches:
+
+1. Implement the code and use your own telephony provider or providers.
+
+2. Outsource the implementation and use your own telephony provider or providers.
+
+3. Use a managed service that manages the process for you.
+
+Some of the main things to consider when choosing an approach are the regions for messages, the expected traffic, the desired reliability, branding requirements, protection from hacking, and your resources.
+
+### Which regions?
+
+Two things can identify a region. First are any regulations for sending messages. Those regulations can be set by collectives, such as the European Union, countries, or even sub-parts of a country. Second is the area covered by the telco sending the message. 
+
+Sending messages to more than one region may have at least two impacts. First, check that your desired vendor or vendors cover those regions.
+
+Second, the features and regulations for traffic may differ from region to region. Some of the differences include:
+
+* Limitations on the types of entities that can send messages by SMS. This typically requires proof of identity and business registration.
+
+* Registration of a sender ID for your business. For example, messages without a valid sender ID are automatically marked as "Likely-SCAM" in Singapore.
+
+* Using *short codes*–special telephone numbers designed for high traffic. This can add significant cost.
```

Comment:
```suggestion
* Using *short codes*—special telephone numbers designed for high traffic. This can add significant cost.
```

Writing style is em dash